### PR TITLE
LPS-190101 Create Date filters from FDSView

### PR DESCRIPTION
**Pull request by fortunatomaldonado**
https://liferay.atlassian.net/browse/LPS-190101

Hello,

Let me know if there are any questions or comments about this.
Thank you!

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java [hunk 1]
```diff
@@ -276,6 +353,27 @@ private String _getAPIURL(
...
+		return JSONUtil.put(
+			"entityFieldType", fdsDateFilterProperties.get("type")
+		).put(
+			"id", fdsDateFilterProperties.get("fieldName")
+		).put(
+			"label", (String)fdsDateFilterProperties.get("label")
+		).put(
+			"max", fdsDateFilterProperties.get("to")
+		).put(
+			"min", fdsDateFilterProperties.get("from")
```
Comment (thektan) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java:371]: According to [DateRangeFilter.js#L201-L205](https://github.com/liferay/liferay-portal/blob/8c68b43be493c0b7ee19908b4844a3c30655635a/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/DateRangeFilter.js#L201-L205), these values should be an object like `{day: '', month: '', year: ''}`

For reference: [DateRangeFDSFilterContextContributor.java#L44-L55](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-impl/src/main/java/com/liferay/frontend/data/set/internal/filter/DateRangeFDSFilterContextContributor.java#L44-L55)

**Review by thektan — CHANGES_REQUESTED** [review]
Hi @fortunatomaldonado!

Through my testing, I was getting a `NoSuchObjectEntryException` so none of the filters were appearing due to this part of the code: https://github.com/liferay-frontend/liferay-portal/pull/3600/files#diff-14ac39b8c7158c0601a7ccd2a2cacb7d24afe76cdd503e7f3c4921a11e3d22c9R167-R170

I left an inline comment for a possible alternative check that could fix the issue.

Also, could these methods be consolidated into a single private method `_getFiltersJSONArray`:
- _getFiltersJSONArray
- getFiltersJSONArray
- getFDSFilters

I felt it was a bit hard to follow the code with so many methods that are only used once. Especially when the purpose of all of them are to build the JSON Array `filters` for the `_reactRenderer.renderReact` call. But you can be the judge to see if combining them makes it more readable or not. At least the `_getFiltersJSONArray` and `getFiltersJSONArray` methods should be combined.

Feel free to reach out if you have any questions!

**Comment by fortunatomaldonado** [comment]
Hello @thektan,

Thank you for taking a look at this!
I am going through your comments and trying to implement them.

Would you be able to provide a video on how you tested the code?
I am not getting some of the values you mentioned and I believe it is due to me not testing correctly.

Truly appreciate the help on this!

**Comment by thektan** [comment]
I essentially created a new Object with a Date field. To add items to the new object, I added it to the Panel App (which I show at the end of the recording). Here's a recording but it's a bit long:

https://github.com/liferay-frontend/liferay-portal/assets/4496722/c82438eb-b519-492c-a5fa-a17c11389d79

**Comment by fortunatomaldonado** [comment]
Hello @thektan,

I updated the code and pushed the changes here.
Let me know if there are any more changes or things to do.

Thank you!

**Review by thektan — APPROVED** [review]
Thanks for making those changes! Looks much cleaner!

**Comment by fortunatomaldonado** [comment]
ci:forward

**Comment by fortunatomaldonado** [comment]
ci:forward

**Comment by mateusralv** [comment]
Hi @fortunatomaldonado the failures are not related! 
[
Testray Comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1692887393879&testrayRunIdA=2411464067&testrayRunIdB=2411511630)
